test/e2e/framework/metrics_util.go

```diff
@@ -621,15 +588,28 @@ func ResetSchedulerMetrics(c clientset.Interface) error {
 	return nil
 }
 
-func convertSampleToBucket(sample *model.Sample, h *histogram, ratio float64) {
-	if sample.Metric["le"] == "+Inf" {
-		h.Buckets[math.MaxFloat64] = int(sample.Value)
-	} else {
-		f, err := strconv.ParseFloat(string(sample.Metric["le"]), 64)
-		if err == nil {
-			h.Buckets[f*ratio] = int(sample.Value)
+func convertSampleToBucket(sample *model.Sample, h *HistogramVec) {
+	labels := make(map[string]string)
+	for k, v := range sample.Metric {
+		if k != "le" {
+			labels[string(k)] = string(v)
+		}
+	}
+	var hist *Histogram
+	for i := range *h {
+		if reflect.DeepEqual(labels, (*h)[i].Labels) {
+			hist = &((*h)[i])
+			break
 		}
 	}
+	if hist == nil {
+		hist = newHistogram(labels)
+		*h = append(*h, *hist)
+	}
+	hist.Buckets[string(sample.Metric["le"])] = int(sample.Value)
+	if sample.Metric["le"] == "+Inf" {
+		hist.Count = int(sample.Value)
```

Comment:
If the `Count` variable is obtained this way, why do you need a separate variable for this? You can just get it directly from `Buckets["+Inf"]`.

Reply:
Also is the assumption that every histogram metric has such a bucket (i.e `le +Inf`) always true?

Reply:
As far as I know, there is always "+Inf" bucket.

Reply:
Ok, sgtm.
We can fix later if this assumption isn't true.

Reply:
I've removed `Count`. We can always use "+Inf".